test/unit/metrics/test_plot_metricframe.py

```diff
@@ -0,0 +1,175 @@
+    return bounds[0]
+
+
+def recall_normal_err(y_true, y_pred):
+    assert len(y_true) == len(y_pred)
+    tn, fp, fn, tp = confusion_matrix(y_true, y_pred).ravel()
+    error = compute_error_metric(tp/(tp+fn), tp + fn, z_score=z_score)
+    return (error, error)
+
+
+def incorrect_negative_recall_normal_err(y_true, y_pred):
+    errors = recall_normal_err(y_true, y_pred)
+    return (-1 * errors[0], -1 * errors[1])
+
+
+def incorrect_single_error_recall_normal_err(y_true, y_pred):
+    errors = recall_normal_err(y_true, y_pred)
+    return errors[0]
+
+
+def accuracy_wilson(y_true, y_pred):
+    assert len(y_true) == len(y_pred)
+    score = accuracy_score(y_true, y_pred)
+    bounds = wilson(score, len(y_true), digits_of_precision, z_score)
+    return bounds
+
+
+def accuracy_normal_err(y_true, y_pred):
+    assert len(y_true) == len(y_pred)
+    score = accuracy_score(y_true, y_pred)
+    error = compute_error_metric(score, len(y_true), z_score)
+    return (error, error)
+
+
+metrics_dict = {
+    'Recall': recall_score,
+    'Recall Bounds': recall_wilson,
+    'Recall Error': recall_normal_err,
+    'Accuracy': accuracy_score,
+    'Accuracy Bounds': accuracy_wilson,
+    'Accuracy Error': accuracy_normal_err,
+    'Recall Bounds Flipped': incorrect_flipped_recall_wilson,
+    'Recall Bounds Single': incorrect_single_bound_recall_wilson,
+    'Recall Error Negative': incorrect_negative_recall_normal_err,
+    'Recall Error Single': incorrect_single_error_recall_normal_err,
+}
+sample_metric_frame = [
+    MetricFrame(
+        metrics=metrics_dict,
+        y_true=y_t,
+        y_pred=y_p,
+        sensitive_features=g_1)
+]
+
+
+@pytest.mark.parametrize("metric_frame", sample_metric_frame)
+def test_plotting_output(metric_frame):
+    axs = plot_metric_frame(metric_frame)
+    assert len(axs) == 4
+    assert isinstance(axs[0], matplotlib.axes.Axes)
+
+
+@pytest.mark.parametrize("metric_frame", sample_metric_frame)
+def test_ambiguous_error_metric(metric_frame):
+    with pytest.raises(ValueError) as exc:
+        plot_metric_frame(metric_frame, metrics=["Recall"], error_bars=[
+                          "Recall Error"], conf_intervals=["Recall Bounds"])
+        assert str(exc.value) == _GIVEN_BOTH_ERROR_BARS_AND_CONF_INT_ERROR
+
+
+@pytest.mark.parametrize("metric_frame", sample_metric_frame)
+def test_unequal_length(metric_frame):
+    with pytest.raises(ValueError) as exc:
+        plot_metric_frame(metric_frame, metrics=[
+                          "Recall", "Accuracy"], error_bars=["Recall Error"])
+        assert str(exc.value) == _METRIC_AND_ERROR_BARS_NOT_SAME_LENGTH_ERROR
+    with pytest.raises(ValueError) as exc:
+        plot_metric_frame(metric_frame, metrics=[
+                          "Recall", "Accuracy"], conf_intervals=["Recall Bounds"])
+        assert str(exc.value) == _METRIC_AND_CONF_INTERVALS_NOT_SAME_LENGTH_ERROR
+
+
+@pytest.mark.parametrize("metric_frame", sample_metric_frame)
```

Comment:
How does this parameterisation work? If `sample_metric_frame` is simply acting as a list of length 1, then I don't think we need the parameterisation here (or on a few other tests)

Reply:
I believe to pass in the MetricFrame into each individual test I need to use parameterization (unless there is another way), which takes in an iterable of parameters. I believe this is intended to make it easy to do tests with multiple sets of parameters, but it is required to be an iterable. I think this is better than calculating the metric frame in each function

Reply:
I think that what you want is a test fixture then. Or for something this simple, you can just call the function to create the sample `MetricFrame` directly from each test. It's only one line.